src/modules/Channel/components/Message/hooks/useIsElementInViewport.ts

```diff
@@ -0,0 +1,20 @@
+import React, { useEffect, useState } from 'react';
+
+export const useIsElementInViewport = (elementRef: React.MutableRefObject<any>) => {
+  const [isVisible, setIsVisible] = useState(false);
+
+  const callback = (entries: IntersectionObserverEntry[]) => {
+    const [entry] = entries;
+    setIsVisible(entry.isIntersecting);
+  };
+
+  useEffect(() => {
+    const observer = new IntersectionObserver(callback);
+    if (elementRef.current) {
+      observer.observe(elementRef.current);
+    }
+    return () => observer.disconnect();
+  }, [elementRef]);
+
+  return { isVisible };
```

Comment:
If we return an object, it'll trigger unnecessary re-rendering where it's being used. 
Unless there's no specific reason we return an object format in here, we'd better just return `isVisible` instead.

Reply:
ah i miss this thanks!